app/models/manageiq/providers/vmware/infra_manager/refresh_parser_inventory_object.rb

```diff
@@ -0,0 +1,1713 @@
+        uids[:hosts],
+        uids[:clusters_by_host],
+        uids[:lans]
+      )
+
+      result[:folders], uids[:folders] = inv_to_ems_folder_hashes(inv)
+      result[:resource_pools], uids[:resource_pools] = rp_inv_to_hashes(inv[:rp])
+
+      result[:customization_specs] = customization_spec_inv_to_hashes(inv[:customization_specs]) if inv.key?(:customization_specs)
+
+      #link_ems_metadata(result, inv)
+      #link_root_folder(result)
+      #set_hidden_folders(result)
+      #set_default_rps(result)
+
+      result
+    end
+
+    def storage_inv_to_hashes(inv)
+      result = add_inventory_collection(Storage, :storages, [:location])
+      result_uids = {:storage_id => {}}
+      return result, result_uids if inv.nil?
+
+      inv.each do |mor, storage_inv|
+        mor = storage_inv['MOR'] # Use the MOR directly from the data since the mor as a key may be corrupt
+
+        summary = storage_inv["summary"]
+        next if summary.nil?
+
+        capability = storage_inv["capability"]
+
+        loc = uid = self.class.normalize_storage_uid(storage_inv)
+
+        new_result = {
+          :ems_ref            => mor,
+          :ems_ref_obj        => mor,
+          :name               => summary["name"],
+          :store_type         => summary["type"].to_s.upcase,
+          :total_space        => summary["capacity"],
+          :free_space         => summary["freeSpace"],
+          :uncommitted        => summary["uncommitted"],
+          :multiplehostaccess => summary["multipleHostAccess"].to_s.downcase == "true",
+          :location           => loc,
+        }
+
+        unless capability.nil?
+          new_result.merge!(
+            :directory_hierarchy_supported => capability['directoryHierarchySupported'].blank? ? nil : capability['directoryHierarchySupported'].to_s.downcase == 'true',
+            :thin_provisioning_supported   => capability['perFileThinProvisioningSupported'].blank? ? nil : capability['perFileThinProvisioningSupported'].to_s.downcase == 'true',
+            :raw_disk_mappings_supported   => capability['rawDiskMappingsSupported'].blank? ? nil : capability['rawDiskMappingsSupported'].to_s.downcase == 'true'
+          )
+        end
+
+        new_result = result.new_inventory_object(new_result)
+
+        result << new_result
+        result_uids[mor] = new_result
+        result_uids[:storage_id][uid] = new_result
+      end
+      return result, result_uids
+    end
+
+    def storage_profile_inv_to_hashes(profile_inv, storage_uids, placement_inv)
+      result = add_inventory_collection(StorageProfile, :storage_profiles)
+      result_uids = {}
+
+      profile_inv.each do |uid, profile|
+        new_result = {
+          :ems_ref                  => uid,
+          :name                     => profile.name,
+          :profile_type             => profile.profileCategory,
+          # TODO: Need to add an InventoryCollection here
+          # TODO: :storage_profile_storages => []
```

Comment:
@Ladas so for something like this how should I create the `InventoryCollection`?  The association is on `StorageProfile` not `ExtManagementSystem` so I don't think creating an `InventoryCollection` with `:parent => @ems` will work right?